presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneMarkDistinctSymbols.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.ConstantProperty;
+import com.facebook.presto.spi.LocalProperty;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.optimizations.StreamPropertyDerivations;
+import com.facebook.presto.sql.planner.optimizations.StreamPropertyDerivations.StreamProperties;
+import com.facebook.presto.sql.planner.plan.MarkDistinctNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.List;
+
+import static com.facebook.presto.sql.planner.plan.Patterns.markDistinct;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class PruneMarkDistinctSymbols
+        implements Rule<MarkDistinctNode>
+{
+    private static final Pattern<MarkDistinctNode> PATTERN = markDistinct()
+            .matching(node -> node.getDistinctSymbols().size() > 1);
+
+    private final Metadata metadata;
+    private final SqlParser sqlParser;
+
+    public PruneMarkDistinctSymbols(Metadata metadata, SqlParser sqlParser)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+    }
+
+    @Override
+    public Pattern<MarkDistinctNode> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Result apply(MarkDistinctNode node, Captures captures, Context context)
+    {
+        List<LocalProperty<Symbol>> localProperties = derivePropertiesRecursively(node.getSource(), context.getLookup(), metadata, context.getSession(), context.getSymbolAllocator().getTypes(), sqlParser)
```

Comment:
Please add a comment to indicate this is a temporary hack and not to be emulated in other rules without careful consideration